Signal/src/ViewControllers/OWSConversationSettingsTableViewController.m

```diff
@@ -158,10 +160,22 @@ - (void)updateEditButton
     }
 }
 
+- (BOOL)hasExistingContact
+{
+    OWSAssert([self.thread isKindOfClass:[TSContactThread class]]);
+
+    TSContactThread *contactThread = (TSContactThread *)self.thread;
+    NSString *recipientId = contactThread.contactIdentifier;
+    SignalAccount *signalAccount = [self.contactsViewHelper signalAccountForRecipientId:recipientId];
+    return signalAccount.contact;
```

Comment:
Done, but we test id/NSObject pointers for nil without explicitly casting them to BOOL in about a million places.